rmw_fastrtps_shared_cpp/test/test_topic_cache.cpp

```diff
@@ -0,0 +1,248 @@
+  EXPECT_EQ(std::count(topic_types.begin(), topic_types.end(), "type1"), 2);
+  EXPECT_EQ(topic_types.at(0), "type1");
+
+  // topic2
+  const auto & it2 = topic_type_map.find("topic2");
+  ASSERT_TRUE(it2 != topic_type_map.end());
+  // Verify that the object returned from the map is indeed the one expected
+  topic_types = it2->second;
+  // Verify that there are two entries for type1
+  EXPECT_TRUE(std::find(topic_types.begin(), topic_types.end(), "type1") != topic_types.end());
+  EXPECT_TRUE(std::find(topic_types.begin(), topic_types.end(), "type2") != topic_types.end());
+}
+
+TEST_F(TopicCacheTestFixture, test_topic_cache_get_participant_map)
+{
+  const auto & participant_topic_map = this->topic_cache.getParticipantToTopics();
+
+  // participant 1
+  const auto & it = participant_topic_map.find(this->guid[0]);
+  ASSERT_TRUE(it != participant_topic_map.end());
+  // Verify that the topic and respective types are present
+  auto & topic_type_map = it->second;
+
+  const auto & topic1it = topic_type_map.find("topic1");
+  ASSERT_TRUE(topic1it != topic_type_map.end());
+  auto topic_types = topic1it->second;
+  EXPECT_TRUE(std::find(topic_types.begin(), topic_types.end(), "type1") != topic_types.end());
+
+  const auto & topic2it = topic_type_map.find("topic2");
+  ASSERT_TRUE(topic2it != topic_type_map.end());
+  topic_types = topic2it->second;
+  EXPECT_TRUE(std::find(topic_types.begin(), topic_types.end(), "type2") != topic_types.end());
+
+  // participant 2
+  const auto & it2 = participant_topic_map.find(this->guid[1]);
+  ASSERT_TRUE(it2 != participant_topic_map.end());
+  // Verify that the topic and respective types are present
+  auto & topic_type_map2 = it2->second;
+
+  const auto & topic1it2 = topic_type_map2.find("topic1");
+  ASSERT_TRUE(topic1it2 != topic_type_map2.end());
+  topic_types = topic1it2->second;
+  EXPECT_TRUE(std::find(topic_types.begin(), topic_types.end(), "type1") != topic_types.end());
+
+  const auto & topic2it2 = topic_type_map2.find("topic2");
+  ASSERT_TRUE(topic2it2 != topic_type_map2.end());
+  topic_types = topic2it2->second;
+  EXPECT_TRUE(std::find(topic_types.begin(), topic_types.end(), "type1") != topic_types.end());
+}
+
+TEST_F(TopicCacheTestFixture, test_topic_cache_get_topic_name_topic_data_map)
+{
+  const auto & topic_data_map = this->topic_cache.getTopicNameToTopicData();
+  auto expected_results = std::map<std::string, std::vector<std::tuple<GUID_t, std::string,
+      rmw_qos_profile_t>>>();
+  expected_results["topic1"].push_back(std::make_tuple(guid[0], "type1", rmw_qos[0]));
+  expected_results["topic1"].push_back(std::make_tuple(guid[1], "type1", rmw_qos[1]));
+  expected_results["topic2"].push_back(std::make_tuple(guid[0], "type2", rmw_qos[0]));
+  expected_results["topic2"].push_back(std::make_tuple(guid[1], "type1", rmw_qos[1]));
+  for (auto & result_it : expected_results) {
+    auto & topic_name = result_it.first;
+    auto & expected_topic_data = result_it.second;
```

Comment:
Was this addressed? I don't see it in the latest diff.

Reply:
Yep, they are addressed [here](https://github.com/ros2/rmw_fastrtps/pull/336/files#diff-047e118adef7162869cfd5f800ece34bR171-R172)

Reply:
`const auto& result_it`

Reply:
Ah! I strangely under the impression that loop vars cannot be const since they get changed after every iteration. Pushed the change!

Reply:
In this case, a `const` iterator means the value referenced by it cannot be changed. Since you only need to read the value, using a `const` is preferable.

Reply:
Yop yop. I've pushed with the change.